### PR TITLE
Fix for #1581 Allow to use podman as container tool for building images

### LINKED ISSUE #1581: Allow to use podman as container tool for building images
**Is your feature request related to a problem? Please describe.**
It's currently not possible to build and deploy operator with the system where only `podman` command is available.

**Describe the solution you'd like**
Rework `Makefile` to allow auto-discovery of container build CLI available on the system from both `docker` or `podman` with an override opption for the user to select tool.

**Describe alternatives you've considered**
Installing podman-docker packages or symlinking podman to docker CLI.

**Additional context**
Operating systems such as Fedora does not have `docker` CLI as the main build, it's easier to select `podman` or make it transparent for the user, so Makefile selects what's available keeping in mind the order that `docker` should be first, `podman` second unless decided by the user to use something specified in the env vars.

### PR DESCRIPTION
**Pull request by mpryc**
The Makefile auto-discovers the build tool, defaulting to docker or podman if available. Users can override this by setting the CONTAINER_TOOL environment variable to specify their preferred tool. This approach is aligned with other operators, such as non-admin, while also adding automatic tool discovery for improved usability and transparency.

## Why the changes were made

Fix #1581 

## How to test the changes made

Reworked Makefile tested on the system with podman and docker.
Selecting CONTAINER_TOOL which is not available on the system properly gave error before entering Makefile parts.

### CODE REVIEW

**Review by mateusoliveira43 — APPROVED** [review]
checked if all places where modified with `grep -Inr "docker" ./Makefile`

Tested with `export CONTAINER_TOOL=batman`

**Comment by mateusoliveira43** [comment]
side note: if we have agreed on https://github.com/openshift/oadp-operator/issues/1540, this change would have already been in the code, as https://github.com/kubernetes-sigs/kubebuilder newer versions already have ability to select `CONTAINER_TOOL` in Makefile

(but would not have the automation to select between docker or podman)

**Review by weshayutin — APPROVED** [review]
ha.. I love seeing the code from the past in new places :)

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,24 @@ AZURE_RESOURCE_FILE ?= /var/run/secrets/ci.openshift.io/multi-stage/metadata.jso
...
+# If neither docker nor podman is found, or if the specified tool is unavailable, the Makefile will exit with an error.
+
+# Set CONTAINER_TOOL to Docker or Podman if not already defined by the user
+CONTAINER_TOOL ?= $(shell \
+  if command -v docker >/dev/null 2>&1; then echo docker; \
+  elif command -v podman >/dev/null 2>&1; then echo podman; \
+  else echo ""; \
+  fi \
+)
+ifeq ($(shell command -v $(CONTAINER_TOOL) >/dev/null 2>&1 && echo found),)
```
Comment (kaovilai) [inline Makefile:40]:
```suggestion
ifeq ($(shell command -v $(CONTAINER_TOOL) >/dev/null 2>&1 && echo $(CONTAINER_TOOL) found),)
```

**Comment by weshayutin** [comment]
https://prow.ci.openshift.org/job-history/gs/test-platform-results/pr-logs/directory/pull-ci-openshift-oadp-operator-master-4.17-e2e-test-kubevirt-aws is broken and needs to be overridden. 

I am not an owner

**Pull request merged**